### ISSUE TITLE
Accuracy and F1

### ISSUE THREAD

**Issue by waiterxiaoyy** [CherBass/ICAM#8]
Hello, I'd like to inquire about the accuracy and F1 score within the training code. Specifically, the original code uses a batch size of 2, so would the accuracy then be limited to 3 values: 0, 0.5, and 1?Also, regarding the F1 score, would it typically produce discrete values like 0.33, 0.66, and 1? Is my understanding correct?

**Comment by ecr05** [comment]
Hi,

I'm not sure I can answer all these questions but given time of year, and the fact Cher has moved to industry, I'm not sure she can reply.

The accuracy would be calculated across the whole training set, how this is passed (i.e batch size) is irrelevant. Everything will get computed after everything has been passed. I don't really know/understand your question about the F1 score.

Emma




Dr Emma C. Robinson

https://metrics-lab.github.io/


Apologies if you get emails outside normal work hours - I have small children and work flexibly. I don't expect a reply at these times.


Senior Lecturer In Medical Image Computing

School of Biomedical Engineering and Imaging Sciences

Kings College London


Office: 5th floor, Becket House, 1 Lambeth Palace Road, London SE1 7EU, UK
Mail: 4th Floor, Lambeth Wing, St Thomas' Hospital, London SE1 7EH
Tel. +44 (0)20 784 89557
________________________________
From: Zhou Yiyang ***@***.***>
Sent: Friday, December 15, 2023 11:54 AM
To: CherBass/ICAM ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [CherBass/ICAM] Accuracy and F1 (Issue #8)


Hello, I'd like to inquire about the accuracy and F1 score within the training code. Specifically, the original code uses a batch size of 2, so would the accuracy then be limited to 3 values: 0, 0.5, and 1?Also, regarding the F1 score, would it typically produce discrete values like 0.33, 0.66, and 1? Is my understanding correct?

**Comment by CherBass** [comment]
Hi,

Thanks for your question. So as Emma mentioned the way metrics are computed is that first predictions are made for the entire dataset, and then metrics are computed, so the batch size is irrelevant in the inference stage. F1 can be a range of 0-1 not discrete.
